src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.cs

```diff
@@ -69,52 +67,42 @@ internal partial class Controller :
             _completionBroker = completionBroker;
         }
 
-        internal static Controller GetInstance(
-            IThreadingContext threadingContext,
-            EditorCommandArgs args,
-            IIntelliSensePresenter<ISignatureHelpPresenterSession, ISignatureHelpSession> presenter,
-            IAsynchronousOperationListener asyncListener,
-            IList<Lazy<ISignatureHelpProvider, OrderableLanguageMetadata>> allProviders,
-            IAsyncCompletionBroker completionBroker)
-        {
-            var textView = args.TextView;
-            var subjectBuffer = args.SubjectBuffer;
-            return textView.GetOrCreatePerSubjectBufferProperty(subjectBuffer, s_controllerPropertyKey,
-                (v, b) => new Controller(threadingContext, v, b,
-                    presenter,
-                    asyncListener,
-                    new DocumentProvider(threadingContext),
-                    allProviders, completionBroker));
-        }
+        public event EventHandler<Model> ModelUpdated;
 
         private SnapshotPoint GetCaretPointInViewBuffer()
         {
             AssertIsForeground();
             return this.TextView.Caret.Position.BufferPosition;
         }
 
-        internal override void OnModelUpdated(Model modelOpt)
+        internal override void OnModelUpdated(Model modelOpt, bool updateController)
         {
             AssertIsForeground();
-            if (modelOpt == null)
-            {
-                this.StopModelComputation();
-            }
-            else
+
+            if (updateController)
             {
-                var triggerSpan = modelOpt.GetCurrentSpanInView(this.TextView.TextSnapshot);
+                if (modelOpt == null)
+                {
+                    this.StopModelComputation();
+                }
```

Comment:
so i'm not sure if i like this, or if it's strictly better.  Is there a case where we might get a modelOpt==null and *not* get updateCOntroller==true?  If so, can we doc that?  If not, can we assert we're not getting that case.  I'm just a little worried about somehow having us null out, but then keeping the controller around with the old value and it all be hinky :)  note: asserts may be sufficient here as opposed to any other code changes.

Reply:
This pull request doesn't change this code flow, so leaving this item for a follow-up (at the earliest, after conversion of this subsystem to NRT).